nikola/plugins/task/archive.py

```diff
@@ -150,10 +152,34 @@ def provide_context_and_uptodate(self, classification, lang, node=None):
             kw["is_feed_stale"] = (datetime.datetime.utcnow().strftime("%Y/%m/%d") != classification)
         else:
             raise Exception("Cannot interpret classification {}!".format(repr(classification)))
+
         context = {
             "title": title,
             "pagekind": [page_kind, "archive_page"],
         }
+
+        # Generate links for hierarchies
+        if kw["create_archive_navigation"]:
+            if hierarchy:
+                # Up level link makes sense only if this is not the top-level
+                # page (hierarchy is empty)
+                context["up_level"] = self.site.link('/'.join(hierarchy[:-1]), lang)
```

Comment:
Your calls to `self.site.link` (this one and the two below) must specify the taxonomy name (here: `archive`).